libs/workspace/src/test-setup.ts

```diff
@@ -13,3 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+jest.mock('fs', () => {
+  const actualFs = jest.requireActual('fs');
+  const unionfs = require('unionfs').default;
+  unionfs.reset = () => {
+    // fss is unionfs' list of overlays
+    unionfs.fss = [actualFs];
+  };
+  return unionfs.use(actualFs);
+});
```

Comment:
Can you please explain why we need unionfs here? :)

Reply:
As stated in the specs - we merge the memes with the actual fs with unionfs so you can still imoprt and read files from your fs but it overlays the memfs. It tries all file operations first on the memfs and if it does not succeed it callbacks to the normal fs is required we cannot use only memfs as the builders need the actual fs to gather files etc.

![image](https://user-images.githubusercontent.com/11156362/82206323-526ec280-9908-11ea-8d0d-0da7518c5053.png)


This enables the usage of adding a memfs on top of the normal fs

```typescript
  const vol = Volume.fromJSON(
    {
      'index.original.html': 'original',
    },
    '/test-root',
  );

  // Merge the current fs with the mocked volume
  const fsMock: any = fs;
  fsMock.use(vol);
```

and we can get all the changes that are performed since this call on the fs.